### PR TITLE
the OOMKilled status is not always in the lastState field of the cont…

### PR DESCRIPTION
**Pull request by aantn**
…ainer status, but is sometimes in the state field (as in the case of job-related pods). We therefore try to look for OOMKilled in both fields

w

### CODE REVIEW

File: playbooks/robusta_playbooks/oom_killer.py [hunk 1]
```diff
@@ -36,9 +44,21 @@ def oom_killer_enricher(event: NodeEvent):
...
+        oom_statuses_from_state = filter(is_state_in_oom_status, pod.status.containerStatuses)
+        for status in oom_statuses_from_state:
+            container_id_to_oom_killed_state[status.state.terminated.containerID] = (status, status.state)
+
+        oom_statuses_from_last_state = filter(is_last_state_in_oom_status, pod.status.containerStatuses)
+        for status in oom_statuses_from_last_state:
+            container_id_to_oom_killed_state[status.lastState.terminated.containerID] = (status, status.lastState)
+
+        for v in container_id_to_oom_killed_state.values():
+            status = v[0]
```
Comment (aantn) [inline playbooks/robusta_playbooks/oom_killer.py:58]: should be `containerID` not `status` right?

can you just make this `for (containerID, state) in container_id_to_oom_killed_state.values()`?

File: playbooks/robusta_playbooks/oom_killer.py [hunk 2]
```diff
@@ -4,14 +4,22 @@
...
-def is_oom_status(status: ContainerStatus):
+def is_last_state_in_oom_status(status: ContainerStatus):
     if not status.lastState:
         return False
     if not status.lastState.terminated:
         return False
     return status.lastState.terminated.reason == "OOMKilled"
 
 
+def is_state_in_oom_status(status: ContainerStatus):
```
Comment (aantn) [inline playbooks/robusta_playbooks/oom_killer.py:15]: can we avoid the code duplication?

**Review by aantn — COMMENTED** [review]
Looks good and I'm merging. I've left some aesthetic comments, so please clean those up whne you have a chance.

File: playbooks/robusta_playbooks/oom_killer.py (hunk 2 above)
Comment (djarv1337) [reply playbooks/robusta_playbooks/oom_killer.py:15]: We can, but I think it makes the code less readable. Are you okay with avoiding this change?

**Pull request merged**